packages/utils/telemetry-utils/src/sampledTelemetryHelper.ts

```diff
@@ -124,19 +146,72 @@ export class SampledTelemetryHelper implements IDisposable {
 		return returnValue;
 	}
 
+	/**
+	 * Accumulates the custom data and sends it to the logger every {@link SampledTelemetryHelper.sampleThreshold} calls.
+	 *
+	 * @param customData -
+	 * A record storing the custom data to be accumulated and eventually logged.
+	 */
+	private accumulateCustomData(customData: Record<TCustomMetrics, number>): void {
+		for (const key of Object.keys(customData) as TCustomMetrics[]) {
+			this.dataSums[key] = (this.dataSums[key] ?? 0) + customData[key];
+			this.dataMaxes[key] = Math.max(
+				this.dataMaxes[key] ?? Number.NEGATIVE_INFINITY,
+				customData[key],
+			);
+		}
+	}
+
+	/**
+	 * Computes average and sets new names for keys.
+	 *
+	 * @param customData - A record storing the custom data that has been accumulated over {@link SampledTelemetryHelper.sampleThreshold} times.
+	 * @param counts - The number of times the {@link SampledTelemetryHelper.measure} has been called.
+	 * @returns A record with the average and maximum values of the custom data.
+	 */
+	private processCustomData(
+		customData: Record<TCustomMetrics, number>,
+		counts: number,
+	): Record<string, number> {
+		const processedCustomData: Record<string, number> = {};
+
+		for (const key of Object.keys(customData) as TCustomMetrics[]) {
+			if (this.dataSums[key] !== undefined) {
+				processedCustomData[`avg${key}`] = roundToDecimalPlaces(
+					this.dataSums[key]! / counts,
+					6,
+				);
+			}
+			if (this.dataMaxes[key] !== undefined) {
+				processedCustomData[`max${key}`] = this.dataMaxes[key] ?? 0; // Added a default value to prevent undefined. But trying to verify why this can be undefined.
```

Comment:
We should probably do something like `max_${key}` so to clearly distinguish the terms in the property name. I.e. `max_age` is better than `maxage`.

Reply:
(Same with "avg" above")